src/orderly.py

```diff
@@ -1,13 +1,103 @@
 from subprocess import run
 
+import os
+import os.path
+import paths
+
 import versions
-from docker_helpers import get_image_name
+from docker_helpers import get_image_name, docker_cp
+from service import orderly_volume_name
+
+from settings import get_secret, save_secret
 from service import orderly_volume_name
 
+orderly_ssh_keypath = ""
 
-def create_empty_store():
-    print("Creating empty Orderly store")
-    image = get_image_name("montagu-reports", versions.reports)
+def create_orderly_store(settings):
+    print("Creating orderly store")
+    image = get_image_name("montagu-orderly", versions.reports)
     run(["docker", "pull", image], check=True)
-    template = "docker run --rm --entrypoint /usr/bin/orderly_init -v {volume}:/orderly {image} /orderly"
-    run(template.format(volume=orderly_volume_name, image=image).split(' '), check=True)
+    if settings["initial_data_source"] == "restore":
+        restore_orderly_store()
+    else:
+        configure_orderly_store(settings)
+
+def restore_orderly_store():
+    print("Restoring orderly permissions")
+    args = ["docker", "run", "--rm", "-d",
+            "--entrypoint", "chmod",
+            "--name", "orderly_setup",
+            "-v", orderly_volume_name + ":/orderly",
+            get_image_name("montagu-orderly", versions.reports),
+            "600", "/orderly/.ssh/id_rsa"]
+    run(args)
+
+def configure_orderly_store(settings):
+    clone = settings['clone_reports']
+    use_real_passwords = settings['use_real_passwords']
+
+    ssh = orderly_prepare_ssh(clone)
+    envir = orderly_prepare_envir(use_real_passwords)
+
+    container = "orderly_setup"
+    args = ["docker", "run", "--rm", "-d",
+            "--entrypoint", "sleep",
+            "--name", container,
+            "-v", orderly_volume_name + ":/orderly",
+            get_image_name("montagu-orderly", versions.reports),
+            "3600"]
+    run(args, check = True)
+
+    try:
+        if clone:
+            print("creating orderly store by cloning montagu-reports")
+            # NOTE: Do this _before_ clone, or we can't do the clone
+            # as credentials are not found
+            docker_cp(ssh, container, "/orderly")
+            docker_cp(envir, container, "/orderly")
+            run(["docker", "exec", container, "montagu-reports-clone"],
+                check = True)
+        else:
+            print("creating empty orderly store")
+            run(["docker", "exec", container, "/usr/bin/orderly_init", "/orderly"],
+                check = True)
+            # NOTE: Do this _after_ init, or we can't do the
+            # initialisation as directory not empty
+            docker_cp(ssh, container, "/orderly")
+            docker_cp(envir, container, "/orderly")
+    finally:
+        paths.delete_safely(paths.orderly)
```

Comment:
Probably better for this line to live in `deploy.py()`, where we delete other sensitive data, for consistency.

Reply:
Fixed in 058178d